### PR TITLE
IGNITE-15889 Add 'contains' method to Record API

### PR DESCRIPTION
**Pull request by JAkutenshi**
https://issues.apache.org/jira/browse/IGNITE-15889

The task was to add `RecordView::contains` method similar to `KeyValueView::contains`. The task requires 4 implementations that similar to `::get` method but without unmarshalling --  only NonNull check of a result of `table::get`.

Also 3 related test classes was updated with contains method testing. Test classes looks a bit different in code style, but I tried to stick to the each certain test class style.

I launched gradle test task, It failed, but the failed test relates to logging module `ClientLoggingTest::testBasicLogging`, I'm not sure should I to fix it or not. Suggestions appreciate.

---

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/api/src/main/java/org/apache/ignite/table/RecordView.java [hunk 1]
```diff
@@ -20,6 +20,8 @@
 import java.util.Collection;
 import java.util.List;
 import java.util.concurrent.CompletableFuture;
+
+import org.apache.ignite.lang.MarshallerException;
```
Comment (SammyVimes) [inline modules/api/src/main/java/org/apache/ignite/table/RecordView.java:24]:
```suggestion
import org.apache.ignite.lang.MarshallerException;
```

File: modules/client/src/test/java/org/apache/ignite/client/ClientRecordViewTest.java [hunk 2]
```diff
@@ -37,6 +37,7 @@
 import java.util.BitSet;
 import java.util.Collection;
 import java.util.List;
+import java.util.Map;
```
Comment (SammyVimes) [inline modules/client/src/test/java/org/apache/ignite/client/ClientRecordViewTest.java:40]:
```suggestion
```

**Review by AMashenkov — APPROVED** [review]

File: modules/table/src/test/java/org/apache/ignite/internal/table/RecordBinaryViewOperationsTest.java [hunk 3]
```diff
@@ -342,6 +342,24 @@ public void getAll() {
...
+        Tuple rec = Tuple.create()
+                .set("id", keyId)
+                .set("val", 11L);
+        Tuple keyRec = Tuple.create()
+                .set("id", keyId);
+
+        tbl.insert(null, rec);
+
+        assertTrue(tbl.contains(null, keyRec));
+        assertFalse(tbl.contains(null, Tuple.create().set("id", -1L)));
```
Comment (AMashenkov) [inline modules/table/src/test/java/org/apache/ignite/internal/table/RecordBinaryViewOperationsTest.java:360]:
```suggestion
        assertTrue(tbl.contains(null, key));
        assertTrue(tbl.contains(null, keyRec));
        assertFalse(tbl.contains(null, Tuple.create().set("id", -1L)));
        assertFalse(tbl.contains(null, Tuple.create().set("id", -1L).set("val", -1L)));
```

Comment (AMashenkov) [reply modules/table/src/test/java/org/apache/ignite/internal/table/RecordBinaryViewOperationsTest.java:360]: Overall looks good. 
Let's add a small fix to make test coverage similar to non-binary version.

Comment (JAkutenshi) [reply modules/table/src/test/java/org/apache/ignite/internal/table/RecordBinaryViewOperationsTest.java:360]: Hm, I tried it without a thought, but later I got that we can't do it: binary record view demands key only tuple as argument. Internally marshaller expected the same kind of input, from `TupleMarshaller.java`:

```java
    /**
     * Marshal tuple key part only.
     *
     * @param tuple Record tuple with key columns only.
     * @return Table row with columns set from given tuples.
     * @throws TupleMarshallerException If failed to marshal tuple.
     */
    Row marshalKey(Tuple tuple) throws TupleMarshallerException;
```

I may guess that there is the strange API situation (in pojo case we can marshall whole pojo with key-value splitting, but for a reasons with tuple marshaller we must pass key-only tuple).

Comment (JAkutenshi) [reply modules/table/src/test/java/org/apache/ignite/internal/table/RecordBinaryViewOperationsTest.java:360]: And I checked: there now no differences with [9627772](https://github.com/apache/ignite-3/pull/3118/commits/9627772996f4cf481f2a9447c8a7aba65dc58378), but check failed. I can't see logs of TC, it shows for me 404 :<

Comment (JAkutenshi) [reply modules/table/src/test/java/org/apache/ignite/internal/table/RecordBinaryViewOperationsTest.java:360]: @AMashenkov should I make any other updates there?

File: modules/table/src/test/java/org/apache/ignite/internal/table/RecordBinaryViewOperationsTest.java [hunk 4]
```diff
@@ -343,6 +343,24 @@ public void getAll() {
...
+        final long keyId = 1L;
+        Tuple rec = Tuple.create()
+                .set("id", keyId)
+                .set("val", 11L);
+        Tuple keyRec = Tuple.create()
+                .set("id", keyId);
+
+        tbl.insert(null, rec);
+
+        assertTrue(tbl.contains(null, keyRec));
```
Comment (AMashenkov) [inline modules/table/src/test/java/org/apache/ignite/internal/table/RecordBinaryViewOperationsTest.java:360]: Let's add tests for cases:
1. `null` value can be found
`tbl.insert(null, Tuple.create().set("id", 1L).set("val", null);`
`assertTrue(tbl.contains(null, keyRec));`
2. removed row can NOT be found
`tbl.delete(null, keyRec);`
`assertFalse(tbl.contains(null, keyRec));`

Comment (JAkutenshi) [reply modules/table/src/test/java/org/apache/ignite/internal/table/RecordBinaryViewOperationsTest.java:359]: Done~

**Review by AMashenkov — APPROVED** [review]

**Pull request merged**